__tests__/butter.test.js

```diff
@@ -1,36 +1,76 @@
-const butter = require('../lib/butter')('api_token');
+import Butter from "../lib/butter.js";
 
-test('should retrive a single page', async () => {
-  const singlePageResponse = await butter.page.retrieve('*', 'example-news-page', {
-    "locale": "en",
-    "preview": 1
-  })
+const butter = Butter("api_token");
 
-  const response = await singlePageResponse.data;
-  const status_code = await singlePageResponse.status;
+test(
+  "should retrieve a single page", 
+  async () => {
 
-  await expect(status_code).toEqual(200);
+    const singlePageResponse = await butter.page.retrieve(
+      "*",
+      "example-news-page", 
+      {
+        "locale": "en",
+        "preview": 1
+      }
+    )
 
-  await expect(response.data.fields.headline).toEqual("This is an example news page");
-  await expect(response.data.slug).toEqual("example-news-page")
-  await expect(response.data.page_type).not.toEqual("sport")
-});
+    const response = await singlePageResponse;
 
-test('should list pages by single-pages', async () => {
-  const singlePageResponse = await butter.page.list('*')
+    await expect(response.data.fields.headline).toEqual("This is an example news page");
+    await expect(response.data.slug).toEqual("example-news-page")
+    await expect(response.data.page_type).not.toEqual("sport")
 
-  const response = await singlePageResponse.data;
-  const status_code = await singlePageResponse.status;
+    return
+  }
+);
 
-  await expect(status_code).toEqual(200);
+test(
+  "should respond with error when access a bad page slug", 
+  async () => {
+    try {
+      const singlePageResponse = await butter.page.retrieve(
+        "*",
+        "fake-post-slug", 
+        {
+          "locale": "en",
+          "preview": 1
+        }
+      )
+    }
+    catch (error) {
+      await expect(error).toBeInstanceOf(Error);
+      await expect(error.message).toEqual("TypeError: Failed to fetch");
+      
+    }
+    
+    return
 
-  await expect(response.meta.count).toEqual(2);
-  await expect(response.data).toHaveLength(2);
+  }
+);
 
-  const firstPage = response.data[0];
+test(
+  "should list pages by single-pages", 
+  async () => {
+    try {
+      const singlePageResponse = await butter.page.list('*')
+      
+      const response = await singlePageResponse;
+  
+      await expect(response.meta.count).toEqual(2);
+      await expect(response.data).toHaveLength(2);
+  
+      const firstPage = response.data[0];
+  
+      await expect(firstPage).toHaveProperty('slug', 'single-page-1');
+      await expect(firstPage).toHaveProperty('fields.title', 'This is a single page');
+  
+      await expect(firstPage).not.toHaveProperty('date_time');
+    }
+    catch (error) {
+      console.error(error)
```

Comment:
Should this lead to test fail?

Reply:
no, it shouldn't. are you getting an error?

![Screen Shot 2024-01-31 at 3 16 20 PM](https://github.com/ButterCMS/buttercms-js/assets/397759/3b70fc22-d0ff-48e6-ac2a-f39d4748e929)

the `try...catch` is how you capture a `Promise.reject` from an await statement. Running the tests all pass for me.

Let me know.

Reply:
The test itself passes, but in case an error happens, the code in the `catch` is run so all `expect`'s are not getting checked. Does `console.error` trigger the test to fail?

Reply:
ah, gotcha. yeah, i'll see what I can do here to update that.

Reply:
@orlyohreally ok, added new test for catch and relay error when page type not found which shows the try/catch in action